src/Symfony/Component/Translation/Translator.php

```diff
@@ -292,6 +292,8 @@ protected function getLoaders()
      */
     public function getMessages($locale = null)
     {
+        trigger_error('The '.__METHOD__.' method is deprecated since version 2.8 and will be removed in 3.0. Use the getCatalogue() method instead.', E_USER_DEPRECATED);
```

Comment:
I would say ``Rely on the TranslatorBagInterface instead``. The current message saying ``Use the getCatalogue() method instead`` is confusing because ``getCatalogue`` is not a drop-in replacement for ``getMessages`` (it is the more powerful way to implement the feature though)